analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -56,8 +58,9 @@ protected SonarAnalysisContextBase(SonarAnalysisContext analysisContext, TContex
     /// <param name="tree">Tree to decide on. Can be null for Symbol-based and Compilation-based scenarios. And we want to analyze those too.</param>
     /// <param name="generatedCodeRecognizer">When set, generated trees are analyzed only when language-specific 'analyzeGeneratedCode' configuration property is also set.</param>
     public bool ShouldAnalyzeTree(SyntaxTree tree, GeneratedCodeRecognizer generatedCodeRecognizer) =>
-        (generatedCodeRecognizer is null || SonarLintFile().AnalyzeGeneratedCode || !tree.IsGenerated(generatedCodeRecognizer, Compilation))
-        && (tree is null || !IsUnchanged(tree));
+        SonarLintFile() is { } sonarLintReader
```

Comment:
If I am not mistaken the `SonarLintFile` Will return a "correct" reader, an empty one or throw an exception. It will never return null, which means there is no need to check for null.

```suggestion
        SonarLintFile() is var sonarLintReader
```